app/src/test/java/com/amaze/filemanager/filesystem/ftpserver/commands/LogMessageFilter.kt

```diff
@@ -0,0 +1,32 @@
+package com.amaze.filemanager.filesystem.ftpserver.commands
+
+import org.apache.ftpserver.ftplet.FtpReply
+import org.apache.mina.core.filterchain.IoFilter
+import org.apache.mina.core.filterchain.IoFilterAdapter
+import org.apache.mina.core.session.IoSession
+import org.apache.mina.core.write.WriteRequest
+
+/**
+ * [IoFilter] to store messages written in an [IoSession]. Test use only.
+ */
+class LogMessageFilter : IoFilterAdapter() {
+    val messages = ArrayList<FtpReply>()
+
+    override fun messageSent(
+        nextFilter: IoFilter.NextFilter?,
+        session: IoSession?,
+        writeRequest: WriteRequest?
+    ) {
+        writeRequest?.message.run {
+            messages.add(this as FtpReply)
```

Comment:
This looks weird, why the cast?

Reply:
Unfortunately [`WriteRequest.getMessage()`](https://nightlies.apache.org/mina/mina/2.1.6/apidocs/org/apache/mina/core/write/WriteRequest.html#getMessage--) returns `Object`.